File: models/doctor.js
const mongoose = require('mongoose')
const Schema = mongoose.Schema

const doctorSchema = new Schema({
    firstName: {
        type: String,
        minlength: 3,
        maxlength: 35,
        required: true
    },
    lastName: {
        type: String,
        minlength: 3,
        maxlength: 20,
        required: true
    },
    picture: {
        type: String,
        required: true
    },
    speciality: {
        type: String,
        minlength: 5,
        maxlength: 20,
        required: true
    },
    phoneNumber: {
        type: String,
        minlength: 9,
        maxlength: 14,
        required: true
    },
    password: {
        type: String,
        minlength: 4,
        maxlength: 4
    }
})

const Doctor = mongoose.model('doctor', doctorSchema)

module.exports = Doctor

File: lib/models/doctor.dart
class Doctor {
  String _id, _firstName, _lastName, _speciality, _phoneNumber, _picture;

  Doctor(
      {String id,
      String firstName,
      String lastName,
      String speciality,
      String phoneNumber,
      String picture})
      : _id = id,
        _firstName = firstName,
        _lastName = lastName,
        _speciality = speciality,
        _phoneNumber = phoneNumber,
        _picture = picture;

  String get picture => _picture;

  set picture(String value) {
    _picture = value;
  }

  String get phoneNumber => _phoneNumber;

  set phoneNumber(String value) {
    _phoneNumber = value;
  }

  String get speciality => _speciality;

  set speciality(String value) {
    _speciality = value;
  }

  String get lastName => _lastName;

  set lastName(String value) {
    _lastName = value;
  }

  String get firstName => _firstName;

  set firstName(String value) {
    _firstName = value;
  }

  String get id => _id;

  set id(String value) {
    _id = value;
  }
}
